### ISSUE TITLE
Streamline listing inclusion

### ISSUE THREAD

**Issue by ibnesayeed** [oduwsdl/wsdlthesis#18]
As per the README:

> [TODO: We need to add a folder for external code samples, streamline the mechanism to add listings, include a list of listings after ToC in the class, and add some helper macros.]

**Comment by machawk1** [comment]
I agree but should we really call this "listings"? I ask in respect to the case of `verbatim` usage in lieu of listings where a text snippet is included that is not code, console readout, etc. Maybe "snippets" but even that is vague.

**Comment by ibnesayeed** [comment]
We can certainly think of a better naming here. :)
